sys/include/net/netopt.h

```diff
@@ -269,10 +269,85 @@ typedef enum {
      */
     NETOPT_LAST_ED_LEVEL,
 
+    /**
+     * @brief   Get/Set preamble length as uint16_t in host byte order.
+     */
+    NETOPT_PREAMBLE_LENGTH,
+
+    /**
+     * @brief   Enable/disable integrity check (e.g CRC).
+     */
+    NETOPT_INTEGRITY_CHECK,
+
+    /**
+     * @brief   Enable/disable channel hopping.
+     */
+    NETOPT_CHANNEL_HOP,
+
+    /**
+     * @brief   Get/Set channel hopping period as uint8_t.
+     */
+    NETOPT_CHANNEL_HOP_PERIOD,
+
+    /**
+      * @brief   Enable/disable single packet reception.
+      * 
+      * If enabled, RX is turned off upon reception of a packet
+      */
+    NETOPT_SINGLE_RECEIVE,
+
+    /**
+     * @brief   Get/Set the reception timeout of a packet.
+     * 
+     * Values are retrieved/passed as uint32_t in host byte order.
+     */
+    NETOPT_RX_TIMEOUT,
+
+    /**
+     * @brief   Get/Set the transmission timeout of a packet.
+     * 
+     * Values are retrieved/passed as uint32_t in host byte order.
+     */
+    NETOPT_TX_TIMEOUT,
+
+    /**
+     * @brief   Get/Set the radio modem type as uint8_t.
+     */
+    NETOPT_DEVICE_MODE,
+
+    /**
+     * @brief   Get/Set the radio modulation bandwidth as uint8_t.
+     */
+    NETOPT_BANDWIDTH,
+
+    /**
+     * @brief   Get/Set the radio spreading factor as uint8_t.
+     */
+    NETOPT_SPREADING_FACTOR,
+
+    /** @brief   Get/Set the radio coding rate as uint8_t.
+     */
+    NETOPT_CODING_RATE,
+
+    /**
+     * @brief   Enable/disable fixed header mode.
+     */
+    NETOPT_FIXED_HEADER,
+
+    /**
+     * @brief   Enable/disable IQ inverted.
+     */
+    NETOPT_IQ_INVERT,
+
+    /**
+     * @brief   Set payload length as uint8_t.
+     */
+    NETOPT_PAYLOAD_LENGTH,
```

Comment:
I removed it since I'm now also convinced it useless in the current state. It can be added later if really required for LoRaWAN and `MAX_PACKET_SIZE` doesn't fit.